src/platform/packages/shared/kbn-standalone-packaging-utils/src/webpack/create_standalone_webpack_config.ts

```diff
@@ -0,0 +1,331 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+/* eslint-disable @kbn/imports/no_boundary_crossing */
+// This is a Node.js-only webpack config generator that imports build-time dependencies
+
+import path from 'path';
+import { CleanWebpackPlugin } from 'clean-webpack-plugin';
+import { BundleAnalyzerPlugin } from 'webpack-bundle-analyzer';
+import { NodeLibsBrowserPlugin } from '@kbn/node-libs-browser-webpack-plugin';
+import sassEmbedded from 'sass-embedded';
+
+export interface StandaloneWebpackOptions {
+  /**
+   * Entry point for the webpack bundle
+   */
+  entry: string;
+
+  /**
+   * Output directory for the built files
+   */
+  outputPath: string;
+
+  /**
+   * Output filename (default: 'index.js')
+   */
+  outputFilename?: string;
+
+  /**
+   * Chunk filename pattern (default: '[name].chunk.js')
+   */
+  chunkFilename?: string;
+
+  /**
+   * External dependencies that won't be bundled
+   * Common externals are automatically included (React, ReactDOM, Monaco, EUI, etc.)
+   */
+  additionalExternals?: Record<string, string> | string[];
+
+  /**
+   * Webpack mode (default: process.env.NODE_ENV || 'development')
+   */
+  mode?: 'development' | 'production';
+
+  /**
+   * Enable bundle analyzer (default: false)
+   */
+  enableBundleAnalyzer?: boolean;
+
+  /**
+   * Path to Kibana root (default: resolves from current directory)
+   */
+  kibanaRoot?: string;
+
+  /**
+   * Path to Babel preset (default: '@kbn/babel-preset/webpack_preset')
+   */
+  babelPreset?: string;
+
+  /**
+   * Additional webpack plugins
+   */
+  additionalPlugins?: any[];
+
+  /**
+   * Clean output directory before build (default: true)
+   */
+  cleanOutputDir?: boolean;
+
+  /**
+   * Enable source maps (default: true)
+   */
+  sourceMaps?: boolean;
+
+  /**
+   * Library target (default: 'commonjs')
+   */
+  libraryTarget?: string;
+}
```

Comment:
nit: We could select properties that exist on the Webpack object config and augment with our own custom requirement